packages/mui-utils/src/resolveProps/resolveProps.ts

```diff
@@ -40,10 +45,21 @@ export default function resolveProps<
               (output[propName] as Record<string, unknown>)[slotPropName] = resolveProps(
                 (defaultSlotProps as Record<string, any>)[slotPropName],
                 (slotProps as Record<string, any>)[slotPropName],
+                mergeClassNameAndStyle,
               );
             }
           }
         }
+      } else if (propName === 'className' && mergeClassNameAndStyle) {
```

Comment:
Check `props.className` so that `clsx` does not run on every iteration.
```suggestion
      } else if (propName === 'className' && mergeClassNameAndStyle && props.className) {
```